game-core/src/main/java/games/strategy/triplea/ui/screen/drawable/TerritoryNameDrawable.java

```diff
@@ -21,6 +21,7 @@
 public class TerritoryNameDrawable implements IDrawable {
   private final String territoryName;
   private final UiContext uiContext;
+  private Rectangle territoryBounds;
```

Comment:
You might want to wrap this in a SoftReference. 
This way it still can be garbage collected when the machine absolutely requires some memory.